hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleContextImpl.java

```diff
@@ -95,22 +96,35 @@ public PreHandleContextImpl(
                 txn,
                 txn.transactionIDOrElse(TransactionID.DEFAULT).accountIDOrElse(AccountID.DEFAULT),
                 configuration,
-                dispatcher);
+                dispatcher,
+                true);
     }
 
-    /** Create a new instance */
     public PreHandleContextImpl(
             @NonNull final ReadableStoreFactory storeFactory,
             @NonNull final TransactionBody txn,
             @NonNull final AccountID payer,
             @NonNull final Configuration configuration,
             @NonNull final TransactionDispatcher dispatcher)
             throws PreCheckException {
+        this(storeFactory, txn, payer, configuration, dispatcher, false);
+    }
+
+    /** Create a new instance */
+    private PreHandleContextImpl(
+            @NonNull final ReadableStoreFactory storeFactory,
+            @NonNull final TransactionBody txn,
+            @NonNull final AccountID payer,
+            @NonNull final Configuration configuration,
+            @NonNull final TransactionDispatcher dispatcher,
+            final boolean isUserTx)
```

Comment:
We should not use a `boolean` here IMO. An enum with two describing names is easier to understand later. Or we can use `TransactionCategory`.

Reply:
@netopyr Makes sense. I will open a PR to fix this once this PR is merged.